packages/icon/src/Icon.spec.tsx

```diff
@@ -154,4 +163,57 @@ describe('Generated glyphs', () => {
   test('Edit icon has displayName: "Edit"', () => {
     expect(EditIcon.displayName).toBe('Edit');
   });
+
+  test('have all been generated', () => {
+    const validGlyphBaseNames: Record<string, true> = {};
+
+    glyphPaths.forEach(filePath => {
+      const baseName = getBaseName(filePath);
+      expect(baseNameToGeneratedFilePath).toHaveProperty(baseName);
+
+      validGlyphBaseNames[baseName] = true;
+    });
+
+    Object.keys(baseNameToGeneratedFilePath).forEach(baseName => {
+      // Make sure there aren't any generated files that don't have a corresponding glyph
+      expect(validGlyphBaseNames).toHaveProperty(baseName);
+    });
+  });
+
+  describe('are up-to-date and not been modified', () => {
+    glyphPaths.forEach(glyphPath => {
+      const baseName = getBaseName(glyphPath);
+
+      test(baseName, () => {
+        const svgFileContents = fs.readFileSync(glyphPath, {
+          encoding: 'utf8',
+        });
+
+        const generatedFileContents = fs.readFileSync(
+          path.resolve(
+            generatedFilesDirectory,
+            baseNameToGeneratedFilePath[baseName],
+          ),
+          {
+            encoding: 'utf8',
+          },
+        );
+
+        const [
+          ,
+          checksum,
+          checkedContents,
+        ] = /^\/\*.*@checksum ([^\n]*)\n.*\*\/\n(.*)$/s.exec(
+          generatedFileContents,
+        )!;
+
+        const expectedChecksum = createHash('md5')
+          .update(svgFileContents)
+          .update(checkedContents)
+          .digest('hex');
+
+        expect(checksum).toEqual(expectedChecksum);
```

Comment:
When this errors can we either inline or log out documentation on how to regenerate the files?